RPerformanceTracking/Private/_RPTTrackingManager.m

```diff
@@ -129,6 +129,7 @@ - (instancetype)init
         } while (0);
         
         [self updateConfiguration];
+        [UIDevice currentDevice].batteryMonitoringEnabled = YES;
```

Comment:
I think we should enable battery monitoring at the end of the above while loop i.e. only enable monitoring if tracking is enabled. And we should disable monitoring if tracking gets turned off.